packages/logging-interceptor/src/logging.interceptor.ts

```diff
@@ -68,13 +122,14 @@ export class LoggingInterceptor implements NestInterceptor {
 
     // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
     const maskedBody = options?.mask?.request ? this.maskData(body, options.mask.request) : body;
+    const maskedHeaders = this.maskHeaders(headers);
 
     this.logger.log(
       {
         message,
         method,
         body: maskedBody,
-        headers,
+        headers: maskedHeaders,
```

Comment:
In case of exception, we could log the error message instead.

Reply:
Not sure to understand your proposition. You mean passing the message of the exception as the value of the header?

Reply:
This always logs the masked headers. I meant we should have an another flow to log the exception occured during the header masking.

Reply:
I proposed an implementation in my last commit. Tell me if you thought about another solution.